crates/ruff_linter/src/rules/flake8_pie/rules/non_unique_enums.rs

```diff
@@ -103,6 +101,17 @@ pub(crate) fn non_unique_enums(checker: &Checker, parent: &Stmt, body: &[Stmt])
     }
 }
 
+fn is_call_to_enum_auto(checker: &Checker, expr: &Expr) -> bool {
+    if let Expr::Call(ast::ExprCall { func, .. }) = expr {
+        checker
+            .semantic()
+            .resolve_qualified_name(func)
+            .is_some_and(|qualified_name| matches!(qualified_name.segments(), ["enum", "auto"]))
+    } else {
+        false
+    }
```

Comment:
This is slightly neater, maybe:

```suggestion
    expr.as_call_expr().map_or(false, |call| {
        checker
            .semantic()
            .resolve_qualified_name(&call.func)
            .is_some_and(|qualified_name| matches!(qualified_name.segments(), ["enum", "auto"]))
    })
```

We could also consider passing only the `SemanticModel` into this function instead of the whole `Checker`, but I don't feel too strongly about that.

Reply:
Thanks for the review! I updated the code in the latest commit. 
I also replaced `Checker` with `SemanticModel` in helper functions because:
1. They only need the `SemanticModel`.
2. We already have a `semantic` variable on the call site, .
3. Since `Checker` can be mutated through shared references, keeping its usage minimal felt like a safer option.